docs/general/naming.md

```diff
@@ -309,4 +315,104 @@ freely. For example, `system.paging.faults` and `system.network.packets`.
+`{metric_name}`, the `{area}.{metric_name}` pattern SHOULD be used.
+
+Examples:
+
+- `http.client.request.duration`
+- `gen_ai.server.request.duration`
+- `messaging.client.sent.messages`
+- `messaging.process.duration` - the term `process` clearly indicates that
+  the metric is reported by the consumer.
+- `kestrel.connection.duration` - here, `kestrel` is the name of the web server,
+  so no additional indication is necessary.
+
+## System-specific naming
+
+**Status**: [Development][DocumentStatus]
+
+### System (project/product/provider) name attribute
+
+Semantic conventions for a certain area are usually applicable to multiple systems
+(projects, providers, products).
+
+For example, database semantic conventions can be used to describe telemetry for a
+broad range of database systems.
+
+Such conventions SHOULD define an attribute to represent system name following
+`{area}.system|provider|protocol.name` pattern.
+
+For example, database conventions include the `db.system.name` attribute.
+
+### Choosing a System Name
+
+When adding new system to the semantic conventions, follow these principles in descending order of priority:
+
+1. The system name SHOULD adhere to the general attribute naming guidelines outlined in this document,
+   as it will be used as a namespace in [system-specific attribute names](#system-specific-attributes).
+
+2. The system name SHOULD unambiguously identify this specific product or project.
+
+   For example, use `gcp.pubsub` or `oracle.db`. Avoid generic names like `pubsub`,
+   which could refer to multiple messaging products, or `oracle` which could refer to
+   multiple Oracle products.
+
+3. The system name SHOULD match the corresponding project or product name in the following cases:
+
+   - Independent projects that do not belong to a specific company, such as Apache Foundation projects like
+     `kafka` or `cassandra`.
+   - Products with names similar to the owning company, such as `mongodb` or `elasticsearch`
+   - Widely recognized products that are popular outside their company's ecosystem.
+     These products often have trademarks without the company name and have
+     their own top-level domain (e.g. `spring` or `mysql`).
+
+4. In other cases, the system name SHOULD be prefixed with the company (organization,
+   division, or group) name. For cloud services, the name SHOULD use the
+   corresponding cloud provider name. For example, use `aws.dynamodb` or `azure.cosmosdb`.
+
+   The company (organization, division, or group) name SHOULD remain consistent
+   across multiple product names in different semantic convention areas.
+
+### System-specific Attributes
+
+When an attribute is specific to a particular system (project, provider, product),
+the corresponding attribute name SHOULD start with the system name following the
+`{system_name}.*.{property}` pattern.
+
+Examples:
+
+- `cassandra.consistency.level` - Describes the consistency level property
```

Comment:
It'd be nice to explain here that if a metric/attribute can be generalized across different implementations/technologies then `db.consistency.level` could be used.

In general if we could collect more examples here to illustrate the different areas/use-cases that would be great.

Reply:
absolutely, this is what I'm trying to do in https://github.com/open-telemetry/semantic-conventions/pull/1707 and would be interested in your input